evalml/pipelines/components/transformers/preprocessing/text_featurizer.py

```diff
@@ -141,17 +140,17 @@ def transform(self, X, y=None):
             {s: "NaturalLanguage" for s in self._text_columns},
         )
         X_lsa = self._lsa.transform(X_ww_altered)
+        X_nlp_primitives.set_index(X_ww.index, inplace=True)
         if any_nans:
             primitive_features = self._get_primitives_provenance(self._features)
             for column, derived_features in primitive_features.items():
-                X_nlp_primitives.loc[nan_mask[column], derived_features] = pd.NA
+                X_nlp_primitives.loc[nan_mask[column], derived_features] = None
 
             lsa_features = self._lsa._get_feature_provenance()
             for column, derived_features in lsa_features.items():
-                X_lsa.loc[nan_mask[column], derived_features] = pd.NA
-
-        X_nlp_primitives.set_index(X_ww.index, inplace=True)
-
+                X_lsa.loc[nan_mask[column], derived_features] = None
+        X_lsa = X_lsa.astype("float64")
```

Comment:
Enforce that the types are 'float64', in case pandas decides to treat `None` similarly to `pd.NA` or `np.nan`. The latter two cause the dtypes to become objects (and thus treats these columns as categorical)

Reply:
IMO we should use `X_lsa.ww.init(logical_types={col: "Double" for col in X_lsa.columns})`. The reason is that if we don't set a ww type explicitly, `ww` will still infer types for the features in `X_lsa`. Right now, that inference works in our favor because ww uses `pd.dtypes.is_float_dtype`. I think the safer thing to do here is to set an explicit woodwork logical type and avoid the type inference.